frontend/js/pages/DetailBattle.js

```diff
@@ -32,25 +33,30 @@ const Icon = styled.img`
 
 class DetailBattle extends Component {
   componentDidMount() {
-    const { setBattle, match, battles } = this.props;
+    const { fetchBattle, match, battles } = this.props;
```

Comment:
This component does not need to receive all the battles, just the specific battle, use a selector for that.